x-pack/plugins/apm/common/rules/apm_rule_types.ts

```diff
@@ -40,6 +40,9 @@ export enum AggregationType {
   P99 = '99th',
 }
 
+export const NUMBER_OF_TIME_SERIES_FOR_PREVIEW_CHART = 5;
+export const INTERVAL_MULTIPLIER_FOR_LOOKBACK = 5;
```

Comment:
Does `INTERVAL_MULTIPLIER_FOR_LOOKBACK` have to be a shared constant? Feels like it should only be a concern for the preview when specifying the lookback (interval size * number of data points)

Reply:
The `INTERVAL_MULTIPLIER_FOR_LOOKBACK` is used both in server and public, so I moved it to common. On the other hand, `NUMBER_OF_TIME_SERIES_FOR_PREVIEW_CHART` is only used in chart preview component, so I moved it now there and renamed as `NUM_SERIES`.

Reply:
The time range is sent from UI e.g. `5m`. Then we need to change it to `25m` on server, and to show label below chart `Last 25 minutes of data`, we need to change it again on client. The `interval` set in rule (e.g. `5m`) is the source of truth. I don't think we should change that, because that is used in rule execution.

Reply:
Did you see this comment?

Reply:
> I will modify the helper function getIntervalAndTimeRange() to return modified start, end and interval.

Hmm.. isn't `getIntervalAndTimeRange` working as intended? You don't need `INTERVAL_MULTIPLIER_FOR_LOOKBACK ` - it's already defined inside `getIntervalAndTimeRange` as `BUCKET_SIZE`

Reply:
Afaict `getIntervalAndTimeRange` provides you with everything the preview chart needs: `start`, `end` and `interval`. What do you need `INTERVAL_MULTIPLIER_FOR_LOOKBACK` for?

Reply:
I thought `getIntervalAndTimeRange` was used elsewhere too, but now I see it is only used for preview chart. So it doesn't need any modifications except `BUCKET_SIZE` which is 20. I will change it to 5. I think  `INTERVAL_MULTIPLIER_FOR_LOOKBACK` or `INTERVAL_MULTIPLIER` is more suitable name than `BUCKET_SIZE`.

Reply:
> I will change it to 5. I think `INTERVAL_MULTIPLIER_FOR_LOOKBACK` or `INTERVAL_MULTIPLIER` is more suitable name than `BUCKET_SIZE`

What about simply `interval` or `fixedInterval` then? That's what it's called in Elasticsearch

Reply:
I will keep the name `BUCKET_SIZE` as is, only changing it to 5. 😄